### ISSUE TITLE
Piggy Banks - Draw Funds from Multiple Accounts

### ISSUE THREAD

**Issue by BugPhobic** [firefly-iii/firefly-iii#6557]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

I LOVE the piggy bank feature, and that is the sole reason I switched to this app.

I have a checking account with a big minimum balance requirement, and a savings account too. My piggy banks are often split between the two accounts.

### Solution

I believe the ability to draw funds from multiple accounts to one piggy bank instead of only one account will be a big improvement.

**Comment by JC5** [comment]
I'll put it on the list!

**Comment by alaexa** [comment]
upvote on this

**Comment by Dan-Sun** [comment]
Yeah, I also have multiple accounts for actually 5 piggy banks. Depending on the purpose, all listed under one group. The description/note doesn't really make sense then either. But that's not much of a problem. I always use the first one for the description.

An example for a new roof:
![new_roof](https://user-images.githubusercontent.com/95150684/206274833-725e5642-0e7e-4bc4-95c3-809a047d2c11.png)

It would be good to create a piggy bank named "New Roof", which can contain several accounts listed under the "Home" group. This way you can easily add more piggy banks under the group "Home", like "New Wire", "Mortgage" ...
The list of all piggy banks would then also be clearer. And the work of creating multiple piggy banks would be less, I guess.

**Comment by Weiming-Hu** [comment]
Being able to draw money from different accounts to the same piggy bank would be so helpful.

**Comment by matthewmorris09** [comment]
Just want to add a quick +1 on this feature. I've been saving for a vacation and happen to also have some gift cards I'll be using it for it. It's really great that I can add something like an increasing gift card as an asset, but now I'm stuck in wanting to include it for a piggy bank.

**Comment by poupouproject** [comment]
Just add Quick +1 on this feature. I have multiple saving account type for one goal.

**Comment by Fmstrat** [comment]
Any reason I should avoid taking this on as a PR? Assuming I should fork from `develop`? Thanks.

**Comment by JC5** [comment]
That's not simple, it requires at least a jump to 6.2.x and there are plenty of other places where piggy banks are used. Feel free to fork the code though 😀

**Comment by diamant-x** [comment]
Interested as well on this feature.

The use case has been shared, but again to explain it myself: Piggy bank savings, being longer-term decision, require money to be stored for long time. Long time stored money is usually diversified across different accounts (either as simple as resilience across two bank entities 'just in case', or for more complex use case such as interests generating). As a result, money for 1 piggy bank ends up split across accounts (for example, housing expenses, or new car).

Just to throw some ideas, from the front-end at least, I would imagine in the piggy bank page, taking use of the two tables view that there's currently there. I can add in the table below, were currently the different used asset accounts are listed, I would be able to search and add additional asset accounts I want to consider its balance towards the 'Sum ob balances for piggy banks'.
Then, in the above table, when CR a piggy bank, I no longer need to select a specific asset account. Instead, all piggy banks checks towards the total sum of balances left for its amount validation.
